### PR TITLE
Add user provider into ResourceProvider

### PR DESCRIPTION
**Pull request by ValdasK**
Adding ability to load user via `UserProviderInterface`, this is needed for related libraries to work (Symfony2 bundle)

### CODE REVIEW

**Comment by hswong3i** [comment]
First of all by default if you hope to protect resource endpoint with HTTP Basic, why don't use silex's 'http' firewall?

Next I guess by adding this we will have a hybrid mode of resource endpoint support, am I correct?

Finally may you also update with corresponding phpunit test case and document, too?

**Comment by ValdasK** [comment]
After quick research, I'v found that by default Silex does not provide this functionality (not sure how to create tests for this), only Symfony does. This change is aimed to extend capabilities of corresponding Symfony bundle.

The goal is to achieve user authentication only using access token. (we are loading users from DB, creating access tokens via Resource Owner Password Credentials Grant).

Currently, when using `oauth2_resource` firewall, created token only has username stored as string, which in real world application is not that exiting. That requires loading user manually each time, and does not provide full integration with framework as one would expect.

**Comment by hswong3i** [comment]
We have [ResourceProviderTest](https://github.com/authbucket/oauth2-php/blob/master/tests/Security/Authentication/Provider/ResourceProviderTest.php) that you can extend for.

You may also have a look with [ResourceType tests](https://github.com/authbucket/oauth2-php/tree/master/tests/ResourceType) and add more test case for your new code.

All such test cases are already build with similar coding style as that in [oauth2-symfony-bundle's tests](https://github.com/authbucket/oauth2-symfony-bundle/tree/master/tests) which you may also reference with. They are 99% equal as that in [oauth2-php's tests](https://github.com/authbucket/oauth2-php/tree/master/tests).

Long story short, I can't understand how critical and useful for adding in your code, please show me with some real case demo, i.e. by extending our existing test case which will trigger your newly added logic ;-)

**Comment by hswong3i** [comment]
Sorry as no more update with test case proof so this pull request now closed.
